breakpoint.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include "map.h"
+#include "riscv.h"
+
+typedef struct {
+    riscv_word_t addr;
+    uint32_t orig_instr;
+} breakpoint_t;
+
+typedef struct {
+    map_t map;
+} breakpoint_map_t;
+
+void breakpoint_map_init(breakpoint_map_t *breakpoint_map);
```

Comment:
How about refining as `breakpoint_map_new`? It would return the allocated object for breakpoint map rather than explicit pre-allocated one.